lib/ansible/modules/storage/netapp/sf_snapshot_schedule_manager.py

```diff
@@ -0,0 +1,451 @@
+       username: "{{ solidfire_username }}"
+       password: "{{ solidfire_password }}"
+       state: present
+       name: Schedule_A
+       time_interval_days: 1
+       starting_date: 2016--12--01T00:00:00Z
+       volumes: 7
+
+   - name: Update Snapshot schedule
+     sf_snapshot_schedule_manager:
+       hostname: "{{ solidfire_hostname }}"
+       username: "{{ solidfire_username }}"
+       password: "{{ solidfire_password }}"
+       state: present
+       schedule_id: 6
+       recurring: True
+       snapshot_name: AnsibleSnapshots
+
+   - name: Delete Snapshot schedule
+     sf_snapshot_schedule_manager:
+       hostname: "{{ solidfire_hostname }}"
+       username: "{{ solidfire_username }}"
+       password: "{{ solidfire_password }}"
+       state: absent
+       schedule_id: 6
+"""
+
+RETURN = """
+msg:
+    description: Successful creation of schedule
+    returned: success
+    type: string
+    sample: '{"changed": true, "key": value}'
+
+msg:
+    description: Successful update of schedule
+    returned: success
+    type: string
+    sample: '{"changed": true}'
+
+msg:
+    description: Successful removal of schedule
+    returned: success
+    type: string
+    sample: '{"changed": true}'
+
+"""
+
+import logging
+from traceback import format_exc
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+import ansible.module_utils.netapp as netapp_utils
+
+HAS_SF_SDK = netapp_utils.has_sf_sdk()
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+class SolidFireSnapShotSchedule(object):
+
+    def __init__(self):
+
+        self.argument_spec = netapp_utils.ontap_sf_host_argument_spec()
+        self.argument_spec.update(dict(
+            state=dict(required=True, choices=['present', 'absent']),
+            name=dict(required=True, type='str'),
+            new_name=dict(required=False, type='str', default=None),
+
+            # interval is required when action == create
+            # interval=dict(choices=['time_interval', 'weekly', 'monthly']),
+
+            time_interval_days=dict(required=False, type='int', default=1),
+            time_interval_hours=dict(required=False, type='int', default=0),
+            time_interval_minutes=dict(required=False, type='int', default=0),
+
+            pause=dict(required=False, type='bool'),
+            recurring=dict(required=False, type='bool'),
+
+            starting_date=dict(type='str'),
+
+            snapshot_name=dict(required=False, type='str'),
+            volumes=dict(required=False, type='list'),
+            retention=dict(required=False, type='str'),
+
+            schedule_id=dict(type='int'),
+        ))
+
+        self.module = AnsibleModule(
+            argument_spec=self.argument_spec,
+            required_if=[
+                ('state', 'present', ['starting_date', 'volumes'])
+            ],
+            supports_check_mode=False
```

Comment:
To me it looks the module does handle check mode (https://github.com/ansible/ansible/pull/20734/files#diff-672a20e0686da08f7554286ee4283346R423). So in my opinion `supports_check_mode` an be set to true